src/append/file.rs

```diff
@@ -84,31 +90,67 @@ impl FileAppenderBuilder {
     }
 
     /// Consumes the `FileAppenderBuilder`, producing a `FileAppender`.
-    /// The path argument can contain environment variables of the form $ENV{name_here},
-    /// where 'name_here' will be the name of the environment variable that
-    /// will be resolved. Note that if the variable fails to resolve,
-    /// $ENV{name_here} will NOT be replaced in the path.
+    /// The path argument can contain special patterns that will be resolved:
+    ///
+    /// - `$ENV{name_here}`: This pattern will be replaced by `name_here`.
+    ///   where 'name_here' will be the name of the environment variable that
+    ///   will be resolved. Note that if the variable fails to resolve,
+    ///   $ENV{name_here} will NOT be replaced in the path.
+    /// - `$TIME{chrono_format}`: This pattern will be replaced by `chrono_format`.
+    ///   where 'chrono_format' will be date/time format from chrono crate. Note
+    ///   that if the chrono_format fails to resolve, $TIME{chrono_format} will
+    ///   NOT be replaced in the path.
     pub fn build<P: AsRef<Path>>(self, path: P) -> io::Result<FileAppender> {
-        let path_cow = path.as_ref().to_string_lossy();
-        let path: PathBuf = expand_env_vars(path_cow).as_ref().into();
-        if let Some(parent) = path.parent() {
+        let new_path = self.date_time_format(path);
+
+        if let Some(parent) = new_path.parent() {
             fs::create_dir_all(parent)?;
         }
         let file = OpenOptions::new()
             .write(true)
             .append(self.append)
             .truncate(!self.append)
             .create(true)
-            .open(&path)?;
+            .open(&new_path)?;
 
         Ok(FileAppender {
-            path,
+            path: new_path,
             file: Mutex::new(SimpleWriter(BufWriter::with_capacity(1024, file))),
             encoder: self
                 .encoder
                 .unwrap_or_else(|| Box::<PatternEncoder>::default()),
         })
     }
+
+    fn date_time_format<P: AsRef<Path>>(&self, path: P) -> PathBuf {
+        let path_cow = path.as_ref().to_string_lossy();
+        // Extract the environment path first
+        let env_path: PathBuf = expand_env_vars(path_cow).as_ref().into();
```

Comment:
Let's keep these two calls separate from each other.  I.e. Lets expect the env then call date_time_format instead of calling expand inside of dtf.

Reply:
I updated to seperate ENV and TIME format. beside that, I also supported multiple placeholders with latest commit